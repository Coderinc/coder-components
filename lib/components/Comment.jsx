import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import moment from 'moment';
import Avatar from '../elements/Avatar';
import Paragraph from '../elements/Paragraph';

const CommentWrapper = styled.div`
  display: flex;
  width: 100%;
  padding: 1rem;
`;

const StyledAvatar = styled(Avatar)`
  margin-top: 0.375rem;
`;

const Text = Paragraph.extend`
  margin: 0 1rem 0 1rem;
  flex: 1;
`;

const Time = Paragraph.extend`
  color: ${({ theme }) => theme.offsetFont};
  font-size: 0.75rem;
  max-width: 5rem;
  text-align: right;
`;

const Comment = ({ user, comment, createdAt }) => (
  <CommentWrapper>
    <StyledAvatar {...user} large />
    <Text>
      {comment}
    </Text>
    <Time>
      {moment(createdAt).calendar()}
    </Time>
  </CommentWrapper>
);

Comment.propTypes = {
  user: PropTypes.object.isRequired,
  comment: PropTypes.string.isRequired,
  createdAt: PropTypes.oneOf([PropTypes.object, PropTypes.string]).isRequired,
};

export default Comment;
