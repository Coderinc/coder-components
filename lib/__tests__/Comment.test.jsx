import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';
import Comment from '../components/Comment';

describe('Comment', () => {
  it('renders correctly', () => {
    const user = { firstName: 'Joe', lastName: 'Schmoe' };
    const comment = 'Lorem Ipsum';
    const component = renderer.create(<Comment user={user} comment={comment} createdAt={(new Date(0)).toISOString()} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});