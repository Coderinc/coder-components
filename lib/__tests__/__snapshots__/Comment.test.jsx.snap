// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment renders correctly 1`] = `
.c4 {
  position: relative;
}

.c6 {
  bottom: -0.375rem;
  left: 50%;
  -webkit-transform: translate(-50%,100%);
  -ms-transform: translate(-50%,100%);
  transform: translate(-50%,100%);
  position: absolute;
  width: 0;
  height: 0;
  z-index: -1;
  opacity: 0;
  -webkit-transition: 0.125s all;
  transition: 0.125s all;
  -webkit-transition-delay: 0.1s;
  transition-delay: 0.1s;
  bottom: 0;
  border-left: 0.375rem solid transparent;
  border-right: 0.375rem solid transparent;
  border-bottom: 0.375rem solid;
}

.c3:hover .c6 {
  opacity: 1;
  z-index: 5;
}

.c5 {
  bottom: -0.375rem;
  left: 50%;
  -webkit-transform: translate(-50%,100%);
  -ms-transform: translate(-50%,100%);
  transform: translate(-50%,100%);
  position: absolute;
  padding: 0.375rem;
  border-radius: 0.125rem;
  font-size: 0.75rem;
  line-height: 0.75rem;
  text-align: center;
  white-space: nowrap;
  box-shadow: 0 1px 3px rgba(0,0,0,0.2),0 1px 2px rgba(0,0,0,0.04);
  z-index: -1;
  opacity: 0;
  -webkit-transition: 0.125s all;
  transition: 0.125s all;
  -webkit-transition-delay: 0.1s;
  transition-delay: 0.1s;
}

.c3:hover .c5 {
  opacity: 1;
  z-index: 5;
}

.c2 {
  width: 2rem;
  height: 2rem;
  border-radius: 50%;
}

.c7 {
  width: 100%;
  height: 100%;
  border-radius: 50%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 100%;
  padding: 1rem;
}

.c1 {
  margin-top: 0.375rem;
}

.c8 {
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: lighter;
  margin: 0 1rem 0 1rem;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c9 {
  font-size: 0.875rem;
  line-height: 1.25rem;
  font-weight: lighter;
  font-size: 0.75rem;
  max-width: 5rem;
  text-align: right;
}

<div
  className="c0"
>
  <div
    className="c1 c2"
  >
    <div
      className="c3 c4"
    >
      <span
        className="c5"
      >
        Joe Schmoe
      </span>
      <div
        className="c6"
      />
      <img
        alt="Joe"
        className="c7"
        src={null}
      />
    </div>
  </div>
  <p
    className="c8"
  >
    Lorem Ipsum
  </p>
  <p
    className="c9"
  >
    12/31/1969
  </p>
</div>
`;
